org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/server/wizard/OrganizationWizardPage.java

```diff
@@ -0,0 +1,140 @@
+    container.setLayoutData(layoutData);
+
+    createOrganizationField(container);
+    setControl(container);
+  }
+
+  private void createOrganizationField(final Composite container) {
+    Label labelOrganization = new Label(container, SWT.NULL);
+    labelOrganization.setText("Organization:");
+    Text organizationText = new Text(container, SWT.BORDER | SWT.SINGLE);
+    GridData gd = new GridData(GridData.FILL_HORIZONTAL);
+    organizationText.setLayoutData(gd);
+
+    DataBindingContext dbc = new DataBindingContext();
+    dbc.bindValue(
+      WidgetProperties.text(SWT.Modify).observe(organizationText),
+      BeanProperties.value(ServerConnectionModel.class, ServerConnectionModel.PROPERTY_ORGANIZATION)
+        .observe(model));
+
+    WizardPageSupport.create(this, dbc);
+
+    ContentProposalAdapter contentProposalAdapter = new ContentAssistCommandAdapter(
+      organizationText,
+      new TextContentAdapter(),
+      new OrganizationProvider(model, this),
+      ITextEditorActionDefinitionIds.CONTENT_ASSIST_PROPOSALS,
+      null,
+      true);
+    contentProposalAdapter.setAutoActivationCharacters(null);
+    contentProposalAdapter.setProposalAcceptanceStyle(ContentProposalAdapter.PROPOSAL_REPLACE);
+    contentProposalAdapter.setFilterStyle(ContentProposalAdapter.FILTER_NONE);
+    contentProposalAdapter.setAutoActivationDelay(100);
+  }
+
+  @Override
+  public void setVisible(boolean visible) {
+    super.setVisible(visible);
+    if (visible) {
+      tryLoadOrganizations();
+      TextSearchIndex<RemoteOrganization> organizationsIndex = model.getOrganizationsIndex();
+      if (organizationsIndex != null && organizationsIndex.size() <= 1) {
+        CustomWizardDialog customWizardDialog = (CustomWizardDialog) getContainer();
+        if (!customWizardDialog.isMovingBackward()) {
+          // Skip organization selection
+          customWizardDialog.showPage(getNextPage());
+          // Ensure that when pressing back on next page we don't return on organization page
+          getNextPage().setPreviousPage(getPreviousPage());
+        }
+      }
+    }
+  }
+
+  private void tryLoadOrganizations() {
+    try {
+      getContainer().run(true, true, new IRunnableWithProgress() {
+
+        @Override
+        public void run(IProgressMonitor monitor) throws InvocationTargetException, InterruptedException {
+          try {
+            model.setOrganizationsIndex(Server.getOrganizationsIndex(model.getServerUrl(), model.getUsername(), model.getPassword(), monitor));
+            setMessage(null);
+          } finally {
+            monitor.done();
+          }
+        }
+      });
+    } catch (InvocationTargetException e) {
+      SonarLintLogger.get().debug("Unable to download organizations", e.getCause());
+      setMessage(e.getCause().getMessage(), IMessageProvider.ERROR);
+      model.setOrganizationsIndex(null);
+    } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2142)